lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -684,6 +694,237 @@ public void resume_publish_queue() {
+                message.action == ProtocolMessage.Action.nack);
+            for (int i = 0; i < messageCount/2; i++) {
+                senderChannel.publish("pending_queued_message_" + i, "Test pending queued messages " + i,
+                    senderCompletion.add());
+                try { Thread.sleep(delay); } catch(InterruptedException e){}
+            }
+
+            final String firstConnectionId = ably.connection.id;
+
+            /* suppress automatic retries by the connection manager and disconnect */
+            try {
+                Method method = ably.connection.connectionManager.getClass().getDeclaredMethod("disconnectAndSuppressRetries");
+                method.setAccessible(true);
+                method.invoke(ably.connection.connectionManager);
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                fail("Unexpected exception in suppressing retries");
+            }
+            connectionWaiter.waitFor(ConnectionState.disconnected);
+            assertEquals("Disconnected state was not reached", ConnectionState.disconnected, ably.connection.state);
+
+            //send some more messages while disconnected
+            for (int i = 0; i < messageCount/2; i++) {
+                senderChannel.publish("queued_message_" + i, "Test pending queued messages " + i,
+                    senderCompletion.add());
+                try { Thread.sleep(delay); } catch(InterruptedException e){}
+            }
+            //now let's unblock the ack nacks and reconnect
+            mockWebsocketFactory.blockReceive(message -> false);
+            /* Wait for the connection to go stale, then reconnect */
+            try {
+                Thread.sleep(waitInDisconnectedState);
+            } catch (InterruptedException e) {
+            }
+            ably.connection.connect();
+            connectionWaiter.waitFor(ConnectionState.connected);
+            assertEquals("Connected state was not reached", ConnectionState.connected, ably.connection.state);
+
+            /* Verify the connection is new */
+            assertNotNull(ably.connection.id);
+            assertNotEquals("Connection has the same id", firstConnectionId, ably.connection.id);
+
+            /* wait for the publish callback to be called.*/
+
+            ErrorInfo[] errors = senderCompletion.waitFor();
+            assertTrue(
+                "Second round of messages (queued) has errors",
+                errors.length == 0
+            );
+
+            messageWaiter.waitFor(messageCount);
+
+            List<Message> received = messageWaiter.receivedMessages;
+            //we need to discard the first 3 - that were sent before disconnection
+            received.remove(0);
+            received.remove(1);
+            received.remove(2);
+            assertEquals(
+                "Received messages",
+                messageCount, received.size()
+            );
```

Comment:
Again, can we have an assertion that the `msgSerial`s are set correctly?

Reply:
That should be the case in newly improved tests as above